### PR TITLE
Hotfix/wp grouping by users

### PR DESCRIPTION
**Pull request by ulferts**
https://community.openproject.org/work_packages/17685

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://hotfix-wp-grouping-by-users.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/models/query.rb [hunk 1]
```diff
@@ -61,11 +67,16 @@ class Query < ActiveRecord::Base
...
+    QueryColumn.new(:assigned_to, sortable: ["#{User.table_name}.lastname",
+                                             "#{User.table_name}.firstname",
+                                             "#{WorkPackage.table_name}.assigned_to_id"],
+                                  groupable: true),
+    QueryColumn.new(:responsible, sortable: ["#{User.table_name}.lastname",
+                                             "#{User.table_name}.firstname",
+                                             "#{WorkPackage.table_name}.responsible_id"],
+                                  groupable: "#{WorkPackage.table_name}.responsible_id",
+                                  join: 'LEFT OUTER JOIN users as responsible ON ' +
+                                        "(#{WorkPackage.table_name}.responsible_id = responsible.id)"),
```
Comment (ulferts) [inline app/models/query.rb:79]: Can't make it shorter without loosing readability or refactoring all the statements.

File: app/models/query.rb [hunk 2]
```diff
@@ -61,11 +67,16 @@ class Query < ActiveRecord::Base
     QueryColumn.new(:priority, :sortable => "#{IssuePriority.table_name}.position", :default_order => 'desc', :groupable => true),
     QueryColumn.new(:subject, :sortable => "#{WorkPackage.table_name}.subject"),
     QueryColumn.new(:author),
-    QueryColumn.new(:assigned_to, :sortable => ["#{User.table_name}.lastname", "#{User.table_name}.firstname", "#{User.table_name}.id"], :groupable => true),
-    QueryColumn.new(:responsible, sortable: ["#{User.table_name}.lastname", "#{User.table_name}.firstname", "#{User.table_name}.id"], groupable: "#{WorkPackage.table_name}.responsible_id", :join => "LEFT OUTER JOIN users as responsible ON (#{WorkPackage.table_name}.responsible_id = responsible.id)"),
-    QueryColumn.new(:updated_at, :sortable => "#{WorkPackage.table_name}.updated_at", :default_order => 'desc'),
-    QueryColumn.new(:category, :sortable => "#{Category.table_name}.name", :groupable => true),
-    QueryColumn.new(:fixed_version, :sortable => ["#{Version.table_name}.effective_date", "#{Version.table_name}.name"], :default_order => 'desc', :groupable => true),
```
Comment (hschink) [inline app/models/query.rb:68]: I just assume it is intended to remove `updated_at`, `category`, and `fixed_version` here.

Comment (ulferts) [inline app/models/query.rb:68]: well spotted, no, it was not. Merge fail.

**Comment by hschink** [comment]
I'm fine with that PR. Whoever likes to merge it, just merge it if Travis is fine with it too.

**Pull request merged**